service-base/src/main/java/org/eclipse/hono/service/amqp/AmqpServiceBase.java

```diff
@@ -385,6 +384,79 @@ protected final ResourceIdentifier getResourceIdentifier(final String address) {
         }
     }
 
+    /**
+     * Handles a request from a client to establish a link for sending messages to this server.
+     * The already established connection must have an authenticated user as principal for doing the authorization check.
+     *
+     * @param con the connection to the client.
+     * @param receiver the receiver created for the link.
+     */
+    protected void handleAuthorizedReceiverOpen(final ProtonConnection con, final ProtonReceiver receiver) {
```

Comment:
We should keep the original method name `handleReceiverOpen` here and make the method `protected` as you did.
We should then also move the test cases from `HonoMessagingTest` that test this method to `AmqpServiceBaseTest` (which does not exist yet) and then simply remove `handleReceiverOpen` from `HonoMessaging`. The same applies to `handleSenderOpen` ...

WDYT?